website/common/script/content/spells.js

```diff
@@ -105,6 +105,9 @@ spells.wizard = {
     target: 'self',
     notes: t('spellWizardFrostNotes'),
     cast (user) {
+      // Check if chilling frost skill has been previously casted or not.
+      // See #12361 for more details.
+      if (user.stats.buffs.streaks === true) throw new NotAuthorized(t('spellAlreadyCast')(user.language));
```

Comment:
should be BadRequest I think?

Reply:
Yes, that's correct. I had earlier thrown a `BadRequest`. But this line is in `website/common/script/content/spells.js` and throwing a `BadRequest` will require and import from `server/libs/errors` ( `import { BadRequest } from '../../../server/libs/errors';)

Won't the sanity check give an error since we are trying to import from the server in the common code? What do you think?

Reply:
You can import a BadRequest error directly from common code like it's done for NotAuthorized at the top of this file

```
import { NotAuthorized } from '../libs/errors';
```

would become 

```
import { NotAuthorized, BadRequest } from '../libs/errors';
```

the `server/libs/errors ` file basically re-exports the errors that were defined in `common/script/libs/errors`

Reply:
Ah, I see. I have made the changes. Thanks for the help, @paglias ! :)

Reply:
Please let me know if there are any other changes. I haven't yet updated my develop branch and merged it with this branch.